src/Catalog/OptionBuilder.php

```diff
@@ -0,0 +1,214 @@
+<?php
+declare(strict_types=1);
+
+namespace TddWizard\Fixtures\Catalog;
+
+use Magento\Catalog\Api\ProductAttributeRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Eav\Api\AttributeOptionManagementInterface;
+use Magento\Eav\Api\Data\AttributeOptionLabelInterface;
+use Magento\Eav\Model\Entity\Attribute\Option as AttributeOption;
+use Magento\TestFramework\Helper\Bootstrap;
+
+/**
+ * Create a source-model option for an attribute.
+ *
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class OptionBuilder
+{
+
+    /**
+     * @var AttributeOptionManagementInterface
+     */
+    private $optionManagement;
+
+    /**
+     * @var AttributeOption
+     */
+    private $option;
+
+    /**
+     * @var AttributeOptionLabelInterface
+     */
+    private $optionLabel;
+
+    /**
+     * @var string
+     */
+    private $attributeCode;
+
+    /**
+     * OptionBuilder constructor.
+     *
+     * @param AttributeOptionManagementInterface $optionManagement
+     * @param AttributeOption $option
+     * @param AttributeOptionLabelInterface $optionLabel
+     * @param string $attributeCode
+     */
+    public function __construct(
+        AttributeOptionManagementInterface $optionManagement,
+        AttributeOption $option,
+        AttributeOptionLabelInterface $optionLabel,
+        string $attributeCode
+    ) {
+        $this->optionManagement = $optionManagement;
+        $this->option = $option;
+        $this->optionLabel = $optionLabel;
+        $this->attributeCode = $attributeCode;
+    }
+
+    /**
+     * Clone the builder.
+     */
+    public function __clone()
+    {
+        $this->option = clone $this->option;
+    }
+
+    /**
+     * Create an option.
+     *
+     * @param string $attributeCode
+     * @return OptionBuilder
+     * @throws \Magento\Framework\Exception\InputException
+     * @throws \Magento\Framework\Exception\StateException
+     */
+    public static function anOption(string $attributeCode): OptionBuilder
```

Comment:
Makes sense to pass the attribute code here, because you will always need options for specific attributes. What do you think about renaming the method to `anOptionFor(string $attributeCode)` to make it more clear?

Reply:
good idea!